src/worker/ingestion/process-event.ts

```diff
@@ -376,17 +390,19 @@ export class EventsProcessor {
             try {
                 await this.db.moveDistinctIds(otherPerson, mergeInto)
             } catch (error) {
-                Sentry.captureException(error, { extra: { mergeIntoPayload, otherPersonPayload } })
-
+                Sentry.captureException(error, {
+                    extra: { mergeInto, mergeIntoDistinctId, otherPerson, otherPersonDistinctId },
+                })
                 failedAttempts++
-                if (failedAttempts === MAX_FAILED_PERSON_MERGE_ATTEMPTS) {
-                    throw error
+
+                // If a person was deleted in between fetching and moveDistinctId, re-run alias to ensure
+                // the updated persons are fetched and merged safely
+                if (error instanceof RaceConditionError && failedAttempts < MAX_FAILED_PERSON_MERGE_ATTEMPTS) {
+                    shouldRetryAliasOperation = true
+                    break
                 }
 
-                // If a person was deleted in between fetching and moveDistinctId,
-                // re-run alias to ensure the updated persons are fetched and
-                // merged safely
-                return await this.alias(otherPersonDistinctId, mergeIntoDistinctId, teamId, false, failedAttempts)
+                throw error
```

Comment:
Q: Is the other try-catch still valid? if not, it makes sense to perhaps remove the loop alltogether.

Reply:
Should still be valid, as a distinct ID can be added in between `moveDistinctIds` and `deletePerson`, in which case we need to move the IDs again so we can delete without an error.